staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go

```diff
@@ -161,7 +166,7 @@ func NewCmdCreateSecretGeneric(f cmdutil.Factory, ioStreams genericiooptions.IOS
 	cmd.Flags().StringSliceVar(&o.FileSources, "from-file", o.FileSources, "Key files can be specified using their file path, in which case a default name will be given to them, or optionally with a name and file path, in which case the given name will be used.  Specifying a directory will iterate each named file in the directory that is a valid secret key.")
 	cmd.Flags().StringArrayVar(&o.LiteralSources, "from-literal", o.LiteralSources, "Specify a key and literal value to insert in secret (i.e. mykey=somevalue)")
 	cmd.Flags().StringSliceVar(&o.EnvFileSources, "from-env-file", o.EnvFileSources, "Specify the path to a file to read lines of key=val pairs to create a secret.")
-	cmd.Flags().StringVar(&o.Type, "type", o.Type, i18n.T("The type of secret to create"))
+	cmd.Flags().StringVar(&o.Type, "type", o.Type, i18n.T("The string value indicate type of secret which is used to facilitate programmatic handling of secret data. Type can be built-in or user defined and vary in terms of validations performed and constraints kubernetes imposed on them. Empty type value indicate Opaque secret. Learn more about built-in types and usage here : https://kubernetes.io/docs/concepts/configuration/secret/#secret-types "))
```

Comment:
This specifically I think needs to be improved by a code change. I see that the fact that the default secret type being Opaque is inherited from the corev1 Secrets type somehow. I think that this should be indicated by properly using the default input for the StringVar function, which here is simply set using o.Type (a string), but here, because in the `NewSecretOptions` does not define this field in the SecretOptions type, the help text only displays that the default value will be an empty string, so I propose that we change what is returned from NewSecretOptions to add the `Opaque` type as the default return. That way the cobra/pflags help texts render those values automatically and clearly, then we update this line to simply read:

`The type of the secret to create. For more information see https://kubernetes.io/docs/concepts/configuration/secret/#secret-types`

Or something similar.

Reply:
so what do you suggest ? should I do the code change in the same PR? 
I'll need to investigate more and raise another PR if that's what we need.

Reply:
I'll also add this suggested change which is in full description for now
in next commit

Reply:
Yeah I think a subsequent PR to make it so the auto-generated help text is accurate would be cleaner. If you need help with the code change PR feel free to reach out to me with any questions and I'll try my best to help answer them or find someone who knows the answer.

Reply:
thanks a lot . i will investigate and raise the PR. this one though will remain seperate

Reply:
Hi @mpuckett159 
Is this you were talking about ?:
```go
func NewSecretOptions(ioStreams genericiooptions.IOStreams) *CreateSecretOptions {
	return &CreateSecretOptions{
		Type: Opaque,
		PrintFlags: genericclioptions.NewPrintFlags("created").WithTypeSetter(scheme.Scheme),
		IOStreams:  ioStreams,
	}
}
```

Reply:
Hi @mpuckett159 , Regarding what we discussed earlier, is this what you were proposing:
```go 
// NewSecretOptions creates a new *CreateSecretOptions with default value
func NewSecretOptions(ioStreams genericiooptions.IOStreams) *CreateSecretOptions {
	return &CreateSecretOptions{
		Type: "Opaque",
		PrintFlags: genericclioptions.NewPrintFlags("created").WithTypeSetter(scheme.Scheme),
		IOStreams:  ioStreams,
	}
}
```
i'll confirm if it works after testing and raise a pr , Just wanted to confirm i understood right.